data/json/items/ammo/weldgas.json

```diff
@@ -15,6 +15,11 @@
     "color": "red",
     "ammo_type" : "weldgas",
     "effects": [ "COOKOFF" ],
-    "flags": [ "NO_DROP", "IRREMOVABLE" ]
+    "flags": [ "NO_DROP", "IRREMOVABLE" ],
+    "explode_in_fire" : true,
+    "explosion" : {
```

Comment:
Having an `explosion` member should automatically imply `explodes_in_fire` which could be achieved by relatively simple logic in `item_factory.cpp`